core/src/test/java/org/infinispan/executors/LimitedExecutorTest.java

```diff
@@ -57,11 +57,13 @@ public void testConcurrencyLimit() throws Exception {
       limitedExecutor.execute(() -> cf2.complete("bla"));
 
       Thread.sleep(10);
+      assertFalse(cf1.isDone());
       assertFalse(cf2.isDone());
 
       latch.countDown();
       assertEquals("bla", cf1.get(10, SECONDS));
       assertEquals("bla", cf2.get(10, SECONDS));
+      eventuallyEquals(0, executor::getActiveCount);
```

Comment:
Do you still need the two threads with this `eventuallyEquals`? I would rather move that on the beginning of the test method, as this line may not be reached upon test failure (and you don't want to fail the next test method).